From: Jan Engelhardt <development@kopano.io>
Date: Tue, 27 Sep 2016 10:16:47 +0200
Subject: build: support gsoap 2.8.36

soap_new_block was renamed, and we still need one change from unsigned
int to int, like commit bp/8.0.1-25-gcf30251 did.
---
 provider/include/proto.h            | 2 +-
 provider/libserver/ECConversion.cpp | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/provider/include/proto.h b/provider/include/proto.h
index ebc5951..f43d3b9 100644
--- a/provider/include/proto.h
+++ b/provider/include/proto.h
@@ -130,7 +130,7 @@ struct propTagArray {
 };
 
 struct entryList {
-	unsigned int __size;
+	int __size;
 	entryId *__ptr;
 };
 
diff --git a/provider/libserver/ECConversion.cpp b/provider/libserver/ECConversion.cpp
index 463754f..e2b858b 100644
--- a/provider/libserver/ECConversion.cpp
+++ b/provider/libserver/ECConversion.cpp
@@ -270,7 +270,11 @@ SOAP_FMAC3 struct entryList52X * SOAP_FMAC4 soap_in_entryList52X(struct soap *so
 		{	soap->error = SOAP_TAG_MISMATCH;
 			if (soap_flag___ptr && soap->error == SOAP_TAG_MISMATCH)
 			{	unsigned int *p;
+#if GSOAP_VERSION >= 20835
+				soap_alloc_block(soap);
+#else
 				soap_new_block(soap);
+#endif
 				for (a->__size = 0; !soap_element_begin_in(soap, "item", 1, type); ++a->__size) {
 					p = (unsigned int *)soap_push_block(soap, NULL, sizeof(unsigned int));
 					soap_default_unsignedInt(soap, p);
